lib/internal/modules/esm/get_format.js

```diff
@@ -53,12 +64,28 @@ function getFileProtocolModuleFormat(url, context, ignoreErrors) {
   if (experimentalSpecifierResolution !== 'node') {
     // Explicit undefined return indicates load hook should rerun format check
     if (ignoreErrors) return undefined;
-    throw new ERR_UNKNOWN_FILE_EXTENSION(ext, fileURLToPath(url));
+    let suggestion = '';
+    if (getPackageType(url) === 'module' && ext === '') {
+      const config = getPackageScopeConfig(url);
+      const fileBasename = basename(filepath);
+      suggestion = 'Extension-less files are disabled inside of ' +
+        '"type":"module" package.json contexts. The package.json file ' +
+        `${config.pjsonPath} caused this "type":"module" context. Try ` +
+        `changing ${filepath} to have a file extension. Note the "bin" ` +
+        'field of package.json can point to a file with an extension E.G. ' +
```

Comment:
```suggestion
        'field of package.json can point to a file with an extension, for example ' +
```

It's "e.g.", lowercase, but especially for non-English-native speakers I find it best to avoid Latin abbreviations like this or "i.e.".